File: app/src/main/java/com/huanchengfly/tieba/api/models/SearchThreadBean.java
package com.huanchengfly.tieba.api.models;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class SearchThreadBean {
    @SerializedName("no")
    private int errorCode;
    @SerializedName("error")
    private String errorMsg;
    private DataBean data;

    public int getErrorCode() {
        return errorCode;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public DataBean getData() {
        return data;
    }

    public class DataBean {
        @SerializedName("has_more")
        private int hasMore;
        @SerializedName("current_page")
        private int currentPage;
        @SerializedName("post_list")
        private List<ThreadInfoBean> postList;

        public int getHasMore() {
            return hasMore;
        }

        public int getCurrentPage() {
            return currentPage;
        }

        public List<ThreadInfoBean> getPostList() {
            return postList;
        }
    }

    public class ThreadInfoBean {
        private String tid;
        private String pid;
        private String title;
        private String content;
        private String time;
        @SerializedName("post_num")
        private String postNum;
        @SerializedName("forum_name")
        private String forumName;
        private UserInfoBean user;
        private int type;

        public String getForumName() {
            return forumName;
        }

        public String getTid() {
            return tid;
        }

        public String getPid() {
            return pid;
        }

        public String getTitle() {
            return title;
        }

        public String getContent() {
            return content;
        }

        public String getTime() {
            return time;
        }

        public String getPostNum() {
            return postNum;
        }

        public UserInfoBean getUser() {
            return user;
        }

        public int getType() {
            return type;
        }
    }

    public class UserInfoBean {
        @SerializedName("user_name")
        private String userName;
        @SerializedName("user_id")
        private String userId;
        private String portrait;

        public String getUserName() {
            return userName;
        }

        public String getUserId() {
            return userId;
        }

        public String getPortrait() {
            return portrait;
        }
    }
}

File: app/src/main/java/com/huanchengfly/tieba/api/models/SearchThreadBean.kt
package com.huanchengfly.tieba.api.models

import com.google.gson.annotations.SerializedName

class SearchThreadBean {
    @SerializedName("no")
    val errorCode: Int? = null

    @SerializedName("error")
    val errorMsg: String? = null
    val data: DataBean? = null

    inner class DataBean {
        @SerializedName("has_more")
        val hasMore: Int? = null

        @SerializedName("current_page")
        val currentPage: Int? = null

        @SerializedName("post_list")
        val postList: List<ThreadInfoBean>? = null

    }

    inner class ThreadInfoBean {
        val tid: String? = null
        val pid: String? = null
        val title: String? = null
        val content: String? = null
        val time: String? = null

        @SerializedName("post_num")
        val postNum: String? = null

        @SerializedName("forum_name")
        val forumName: String? = null
        val user: UserInfoBean? = null
        val type: Int? = null

    }

    inner class UserInfoBean {
        @SerializedName("user_name")
        val userName: String? = null

        @SerializedName("user_id")
        val userId: String? = null
        val portrait: String? = null

    }
}